sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventProcessorClient.java

```diff
@@ -90,7 +89,7 @@ public class EventProcessorClient {
         EventHubAsyncClient eventHubAsyncClient = eventHubClientBuilder.buildAsyncClient();
 
         this.checkpointStore = Objects.requireNonNull(checkpointStore, "checkpointStore cannot be null");
-        this.identifier = UUID.randomUUID().toString();
+        this.identifier = eventHubAsyncClient.getIdentifier();
```

Comment:
This can potentially be `null` resulting in a `null` identifier for processor which will not work. Also, note that the processor requires the id to be unique across all instances for load balancer to function properly.

Reply:
Sorry, I misunderstand before. I just get confirm with Jesse. Now, if user didn't set an identifier, it will genenrate when client build, so here will never be null. And in each partition pumper, if user doesn't set an identifier in option, these partition pumper will be different from each other. If user set a specific identifier in options pass to processor client builder, these client will use the specific identifier.

Reply:
@ZejiaJiang Do you know how user would use the identifier in partition pumper, if they do not provide in AmqpClientOptions?

Reply:
@weidongxu-microsoft Users don't use identifier directly no matter they provide identifier or not. Identifier are mainly used for creating links.

Reply:
If user does not use the identifier, why provide `getIdentifier()` in e.g. EventHubConsumerClient and EventHubProducerClient?

Reply:
We provide `getIdentifier()` just align with .NET SDK.